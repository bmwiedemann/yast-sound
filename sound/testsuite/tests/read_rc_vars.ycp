{
    include "testsuite.ycp";
    include "sound/routines.ycp";

    TEST(``(read_rc_vars(["ASD"])), [$["rc":$["system":$["var1": "val1", "var2": "val2"]]]], nil);
    TEST(``(read_rc_vars(["var1"])), [$["rc":$["system":$["var1": "val1", "var2": "val2"]]]], nil);
    TEST(``(read_rc_vars(["var1", "var2"])), [$["rc": $["system": $["var1": "val1", "var2": "val2"]]]], nil);
    TEST(``(read_rc_vars([])), [$["rc": $["system": $["var1": "val1", "var2": "val2"]]]], nil);
    TEST(``(read_rc_vars(["var3"])), [$["rc":$["system": $["var1": "val1", "var2": "val2"]]]], nil);
    
    
}
