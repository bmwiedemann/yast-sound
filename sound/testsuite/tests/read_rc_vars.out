Read	.rc.system.ASD nil
Return	$["ASD":nil]
Read	.rc.system.var1 "val1"
Return	$["var1":"val1"]
Read	.rc.system.var1 "val1"
Read	.rc.system.var2 "val2"
Return	$["var1":"val1", "var2":"val2"]
Return	$[]
Read	.rc.system.var3 nil
Return	$["var3":nil]
